views-react/src/main/java/io/micronaut/views/react/ReactJSContext.java

```diff
@@ -15,29 +15,27 @@
  */
 package io.micronaut.views.react;
 
-import io.micronaut.context.annotation.Parameter;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.Qualifier;
+import io.micronaut.context.annotation.Bean;
 import io.micronaut.core.annotation.Internal;
-import io.micronaut.core.annotation.NonNull;
+import io.micronaut.inject.BeanType;
 import jakarta.annotation.PostConstruct;
 import jakarta.annotation.PreDestroy;
 import jakarta.inject.Inject;
-import org.graalvm.polyglot.Context;
-import org.graalvm.polyglot.Source;
-import org.graalvm.polyglot.Value;
+import jakarta.inject.Named;
+import org.graalvm.polyglot.*;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.List;
-
-import static java.nio.charset.StandardCharsets.UTF_8;
+import java.util.stream.Stream;
 
 /**
  * A bean that handles the Javascript {@link Context} object representing a loaded execution
  * environment usable by one thread at a time.
  */
 @Internal
-class JSContext implements AutoCloseable {
+@Bean
```

Comment:
should we `@Singleton` instead of `@Bean`?

Reply:
No, it's not a singleton object. There needs to be multiple of them to enable concurrency.